WordPress/gutenberg
# Block Library: Standardize reduced motion handling with media queries

Part of: #68282 


## What?
Refactors animation and transition styles to use `@media not (prefers-reduced-motion)` ensuring better accessibility for users who prefer reduced motion.

## Why?
Currently, many places in the codebase do not account for user's motion preferences. This might not be correct for user's having reduced motion preferences. This PR fixes that. 

## Testing Instructions
1. Open the Block Editor
2. Add an Image block and upload an image
3. Enable the lightbox feature in block settings
4. Test normal behavior:
   - Click to open Lightbox
   - Observe opening/closing animations
   - Test zoom functionality

5. Test with reduced motion preference:
   - Enable "Reduced motion" preference:
     - Chrome DevTools: Open DevTools > More Tools > Rendering > "Emulate CSS media feature prefers-reduced-motion" > select "reduce"
   - Repeat the above tests
   - Verify that:
     - Lightbox opens/closes without animation
     - No zoom animations occur

6. Disable reduced motion and verify all original animations work correctly

## Screencast 

Image Block - 

https://github.com/user-attachments/assets/a94e75bf-d1ea-4d89-91cb-04a767e27dea

Navigation Block -

https://github.com/user-attachments/assets/10a96c03-e358-448f-8201-3a2f7b4ac8cb


Social Links - 

https://github.com/user-attachments/assets/573674dd-cff1-441d-8f82-129c3c956d7c
